src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.Runtime.cs

```diff
@@ -57,6 +60,65 @@ public sealed override IEnumerable<RuntimeTypeHandle> TryGetImplementedInterface
             return RuntimeAugments.TryGetImplementedInterfaces(typeHandle);
         }
 
+        public sealed override void VerifyInterfaceIsImplemented(RuntimeTypeHandle typeHandle, RuntimeTypeHandle ifaceHandle)
+        {
+            if (RuntimeAugments.IsInterface(typeHandle))
+            {
+                throw new ArgumentException(SR.Argument_InterfaceMap);
+            }
+
+            if (!RuntimeAugments.IsInterface(ifaceHandle))
+            {
+                throw new ArgumentException(SR.Arg_MustBeInterface);
+            }
+
+            foreach (RuntimeTypeHandle iface in RuntimeAugments.TryGetImplementedInterfaces(typeHandle))
```

Comment:
RuntimeAugments.IsAssignableFrom might be better because it also covers variance:

```csharp
interface IFoo<in T>
{
    void Frob();
}



class Foo : IFoo<object>
{
    public void Frob() { }
}

class Program
{
    static void Main()
    {
        var m = typeof(Foo).GetInterfaceMap(typeof(IFoo<string>));
        Console.WriteLine(m.InterfaceMethods[0].DeclaringType);
        Console.WriteLine(m.TargetMethods[0].DeclaringType);
    }
}
```

Reply:
On the other hand, it bails on
```csharp
interface IFoo<in T>
{
    void Frob();
}

class Foo<T> : IFoo<object>
{
    public void Frob() { }
}

class Program
{
    static void Main()
    {
        var m = typeof(Foo<>).GetInterfaceMap(typeof(IFoo<string>));
        Console.WriteLine(m.InterfaceMethods[0].DeclaringType);
        Console.WriteLine(m.TargetMethods[0].DeclaringType);
    }
}
```
due to https://github.com/dotnet/corert/blob/0185f4706ae47cf7397a7e24317fa45ef35edea4/src/Runtime.Base/src/System/Runtime/TypeCast.cs#L493-L497
while .NET Core does not.

Reply:
Could you please add a comment that lists all known cases that CoreCLR handles, but this implementation does not?